### PR TITLE
feat: replace worker file from outside

### PR DESCRIPTION
**Pull request by czy88840616**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,19 +3,9 @@ language: node_js
...
-  - EGG_VERSION=2
-matrix:
-  exclude:
-  - node_js: '6'
-    env: EGG_VERSION=2
 install:
-  - npm i npminstall
-  - sed -i.bak '/"egg":/d' package.json
-  - npminstall -d
+  - npm i npminstall && npminstall
```
Comment (atian25) [inline .travis.yml]: need to revert this file, it's used to test egg1 and egg2

File: lib/master.js [hunk 2]
```diff
@@ -530,6 +528,14 @@ class Master extends EventEmitter {
       process.exit(0);
     }, timeout);
   }
+
+  getAgentWorkerFile() {
```
Comment (atian25) [inline lib/master.js]: add some test case for how to use these ?

**Comment by czy88840616** [comment]
我发现 `mm.cluster` 里没留口子。。没法测了

**Comment by atian25** [comment]
ping @popomore

**Comment by atian25** [comment]
没太看懂这个最终是怎么使用法？
是在上层框架里面去覆盖？这样的话，需要在 egg-cluster 里面把 Master exports 下，然后在 egg 里面也要 exports

**Comment by popomore** [comment]
不用，这个只是内部场景。

我先合并，我来加用例

**Comment by popomore** [comment]
最近 synk 有点问题

**Pull request merged**